### PR TITLE
Initial Setup

### PR DESCRIPTION
**Pull request by agsuy**
- First readme
- Build flags
- Clones and builds any branch

### CODE REVIEW

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+FROM node:14.17-bullseye
+# Install packages
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y \
+    git \
+    jq \
+    sudo \
+    vim
```
Comment (matibat) [inline dockerfile:8]: Maybe `net-tools` is helpful here. I think it adds `ping` and `ifconfig`

File: scripts/gitClone.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#/bin/sh
+
+git clone --depth 1 --branch ${BRANCH:-master} https://github.com/ardriveapp/ardrive-cli.git
```
Comment (matibat) [inline scripts/gitClone.sh:3]: `${BRANCH:-master}`
Does the substitution have a default for empty values? AWESOME! :clap:

File: README.md [hunk 3]
```diff
@@ -1,2 +1,93 @@
...
+
+Sample command:
+
+``docker run --name ardrive --rm --init -it -e BRANCH='dev' --mount type=tmpfs,destination=/home/node/tmp ardrive-cli``
+
+Environment variables that control the build process:
+
+``-e NO_SETUP=1 `` skips cloning and building
+
+``-e NO_AUTOBUILD=1 `` skips building process
```
Comment (matibat) [inline README.md:62]: Beware if you set `NO_AUTOBUILD=0` it takes the same effect as setting it to one

File: dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#Clear apt cache   
+RUN apt-get clean -y
+
+#Set password and sudo
+ARG PASSWORD=ardrive
+RUN echo "root:${PASSWORD}" | chpasswd
+RUN echo "node:${PASSWORD}" | chpasswd
+RUN adduser node sudo
```
Comment (matibat) [inline dockerfile:17]: Are we setting the password before creating the user? Does it work well?

File: dockerfile [hunk 5]
```diff
@@ -0,0 +1,42 @@
+FROM node:14.17-bullseye
+# Install packages
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y \
+    git \
+    jq \
+    sudo \
+    vim
+
+#Clear apt cache
```
Comment (matibat) [inline dockerfile:10]:
```suggestion
# Clear apt cache
```

File: dockerfile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+RUN apt-get install --no-install-recommends -y \
+    git \
+    jq \
+    sudo \
+    vim
+
+#Clear apt cache   
+RUN apt-get clean -y
+
+#Set password and sudo
```
Comment (matibat) [inline dockerfile:13]:
```suggestion
# Set password and sudo
```

File: dockerfile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+ARG PASSWORD=ardrive
+RUN echo "root:${PASSWORD}" | chpasswd
+RUN echo "node:${PASSWORD}" | chpasswd
+RUN adduser node sudo
+#Set home var
```
Comment (matibat) [inline dockerfile:18]:
```suggestion
# Set home var
```

File: scripts/entry.sh [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#/bin/sh
+
+#One Script to rule them all, One Script to find them,
+#One Script to bring them all, and in the arweave bind them.
```
Comment (matibat) [inline scripts/entry.sh:4]: I didn't actually understand this comment :thinking: 
```suggestion
# One Script to rule them all, One Script to find them,
# one Script to bring them all, and in the arweave bind them.
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,93 @@
...
+# Ardrive Cli Docker
+Contains Dockerfile and all CI-related tooling to build a Docker image for the ArDrive CLI.
+
+Knowledge on how to set-up Docker is expected.
+
+Password for both root and node user inside the Docker is "ardrive"
+## Intended Audience
+This tool is intended for use by:
+
+ArDrive power users with advanced workflows and automation in mind: bulk uploaders, those with a larger storage demand, developers, nft creators, storage/db admins, etc.
```
Comment (matibat) [inline README.md:10]:
```suggestion
ArDrive power users with advanced workflows and automation in mind: bulk uploaders, those with a larger storage demand, developers, NFT creators, storage/DB admins, etc.
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,93 @@
...
+## Interact with a wallet
+
+### Put wallet inside your container
+
+To copy our wallet inside Docker, we just need the following command.
+Image was intended to work with only ONE wallet at a time. 
+
+Running the below command a 2nd time will overwrite the 1st wallet.
+
+``docker exec -i ardrive sh -c 'cat > /home/node/tmp/wallet.json' < [path to my wallet file]``
```
Comment (matibat) [inline README.md:73]: Is it more clear? What do you think?
```suggestion
``docker exec -i ardrive sh -c 'cat [path to my wallet file] > /home/node/tmp/wallet.json'``
```

**Review by matibat — COMMENTED** [review]
I left some comments and suggestions.

Question: doesn't the `Dockerfile` need a capital `D` in the file name?

File: README.md (hunk 3 above)
Comment (agsuy) [reply README.md:62]: Yeah u could set it to any non-null value actually. Just used 1 for the example

File: dockerfile (hunk 4 above)
Comment (agsuy) [reply dockerfile:17]: No. We change default node user pass. And we add it to the sudo group

File: scripts/entry.sh (hunk 8 above)
Comment (agsuy) [reply scripts/entry.sh:4]: Easter egg. LOTR reference.

File: README.md (hunk 10 above)
Comment (agsuy) [reply README.md:73]: I would keep it as it is. Also yours does not work as intended.

File: dockerfile (hunk 1 above)
Comment (agsuy) [reply dockerfile:8]: Gotta look into this

File: dockerfile (hunk 5 above)
Comment (agsuy) [reply dockerfile:10]: ll look into comments indentation later.

File: dockerfile [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+RUN chmod +x /home/node/entry.sh
+COPY scripts/.bashrc_patch /home/node/
+RUN chmod +r /home/node/.bashrc_patch
+
+# Use non-root user
+USER node
+WORKDIR $HOME
+# Set env to tmpfs path
+ENV WALLET /home/node/tmp/wallet.json
+# Create uplads folder
```
Comment (arielmelendez) [inline dockerfile:35]: nit: typo

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**